docs/testplan/BGP-Scale-Test.md

```diff
@@ -0,0 +1,121 @@
+
+# Route Configuration Setup
+The count of routes from BGP peers is vital, we will leverage exabpg to advertise routes to all BGP peers, and those routes be be advertised to device under test finally.
+
+When DUT is T0, via exabgp, we will advertise 511 routes with prefix length 120 and 511 rotues with prefix length 128 to each neighbor T1 devices. The prefixes with length 120 are mocking VLAN address on downstream T0s, and the prefixes with length 128 are mocking loopback address on downstream T0s.
+
+When DUT is T1, via exabgp, we will advertise 1 route with prefix 120 to each neighbor T0 devices and every prefix in route is unique.
+
+Detail route scale is described in below table:
+| Topology Type                              | BGP Routes Count      | BGP Nexthop Group Count | BGP Nexthop Group Members Count |
+| ------------------------------------------ | --------------------- | ----------------------- | ------------------------------- |
+| t0-isolated-d2u254s1, t0-isolated-d2u254s2 |  254 * ( 511 + 511 )  | 254                     | 254                             |
+| t0-isolated-d2u510                         |  510 * ( 511 + 511 )  | 510                     | 510                             |
+| t1-isolated-d2u254s1, t1-isolated-d2u254s2 |  254 * ( 1 + 1 )      | 254                     | 1                               |
+| t1-isolated-d2u510                         |   510 * ( 1 + 1 )     | 510                     | 1                               |
+
+
+# Test Cases
+
+## Initial State
+All bgp sessions are up and established and all routes are stable with expected count.
+
+## BGP Sessions Flapping Test
+### Objective
+When BGP sessions are flapping, make sure control plane is functional and data plane has no downtime or acceptable downtime.
+### Steps
+1. Start and keep sending packets with all routes to the random one open port via ptf.
+1. Shutdown one or half random port(s) that establishing bgp sessions. (shut down T1 sessions ports on T0 DUT, shut down T0 sesssions ports on T1 DUT.)
+1. Wait for routes are stable, check if all nexthops connecting the shut down ports are disappeared in routes.
+1. Stop packet sending
+1. Estamite data plane down time by check packet count sent, received and duration.
+
+
+## Unisolation Test
+### Objective
+In the worst senario, verify control/data plane have acceptable conergence time.
+### Steps
+1. Shut down all ports on device. (shut down T1 sessions ports on T0 DUT, shut down T0 sesssions ports on T1 DUT.)
+1. Wait for routes are stable.
+1. Start and keep sending packets with all routes to all portes via ptf.
+1. Unshut all ports and wait for routes are stable.
+1. Stop sending packets.
+1. Estamite control/data plane convergence time.
+
+
+## Nexthop Group Member Scale Test
+### Objective
+When routes on BGP peers are flapping, make sure DUT's control plane is functional and data plane has no downtime or acceptable downtime.
```

Comment:
The objective should be make sure the large next hop group member is supported by the control plane and data plane within acceptable downtime.